File: ks1.py
def sumpair(t,s):
    su = 0
    if s == 1:
        return 0
    else:
        for i in range(s):
            su += i*t[i] - (s-i-1)*t[i]
        return su

for _ in range(int(input())):
    n = int(input())
    A = list(map(int, input().split()))
    pre_xor = [A[0]]
    for i in range(1, n):
        pre_xor.append(A[i]^pre_xor[i-1])
    dp = {}
    ans = 0
    for i in range(n):
        if pre_xor[i] == 0:
            ans += i
        if pre_xor[i] in dp.keys():
            dp[pre_xor[i]].append(i)
        else:
            dp[pre_xor[i]] = [i]
    for x in dp.values():
        s = len(x)
        t = sumpair(x,s)
        if t != 0:
            ans += t - s*(s-1)//2
    print(ans)

File: ks1.cpp
#include <iostream>
#include <vector>
#include <map>

using namespace std;

struct indexes{
    vector<int> val;
};

int main(){
    int t;cin>>t;
    for(int i=0;i<t;i++){
        int n;cin>>n;
        vector<int> A;
        for(int a=0;a<n;a++){
            int tmp;cin>>tmp;
            A.push_back(tmp);
        }
        vector<int> pre_xor;
        pre_xor.push_back(A[0]);
        for(int a=1;a<n;a++){
            pre_xor.push_back(pre_xor[a-1]^A[a]);
        }
        int ans = 0;
        map<int, vector<int>> dp;
        for(int c=0;c<n;c++){
            if(dp.find(pre_xor[c]) != dp.end()){
                for(int d=0;d < dp[pre_xor[c]].size();d++){
                    ans += c-dp[pre_xor[c]][d]-1;
                }
            }
            if(pre_xor[c] == 0){
                ans += c;
            }
            dp[pre_xor[i]].push_back(c);
        }
        cout<<ans<<"\n";
    }
    return 0;
}